cvat/apps/engine/serializers.py

```diff
@@ -27,6 +27,70 @@
 
 from cvat.apps.engine.view_utils import build_field_filter_params, get_list_view_name, reverse
 
+
+class WriteOnceMixin:
+    """
+    Adds support for write once fields to serializers.
+
+    To use it, specify a list of fields as `write_once_fields` on the
+    serializer's Meta:
+    ```
+    class Meta:
+        model = SomeModel
+        fields = '__all__'
+        write_once_fields = ('collection', )
+    ```
+
+    Now the fields in `write_once_fields` can be set during POST (create),
+    but cannot be changed afterwards via PUT or PATCH (update).
+    Inspired by http://stackoverflow.com/a/37487134/627411.
+    """
+
+    def get_extra_kwargs(self):
+        extra_kwargs = super().get_extra_kwargs()
+
+        # We're only interested in PATCH/PUT.
+        if 'update' in getattr(self.context.get('view'), 'action', ''):
+            extra_kwargs = self._set_write_once_fields(extra_kwargs)
+
+        return extra_kwargs
+
+    def get_fields(self):
```

Comment:
Is there any point in overriding both `get_extra_kwargs` and `get_fields`? Seems like you could just remove the intermediate step (setting the `read_only` kwarg) and just use the `write_once_fields` setting directly in this function.

Reply:
Technically, leaving just the `get_extra_kwargs` would be ideal, but it only affects the fields from the model, not the manually defined ones.

Reply:
It would, but since it doesn't work - by design - I don't think we should persist with the kwargs approach.